pkg/kev/manifest.go

```diff
@@ -181,39 +188,83 @@ func (m *Manifest) ReconcileConfig(envs ...string) (*Manifest, error) {
 	if err != nil {
 		sg := m.UI.StepGroup()
 		defer sg.Done()
-		renderStepError(m.UI, sg.Add(""), renderStepReconcile, err)
+		renderStepError(m.UI, sg.Add(""), renderStepReconcileDetect, err)
 		return nil, err
 	}
 
 	for _, e := range filteredEnvs {
-		if err := validateExtensions(e.override); err != nil {
+		if err := validateEnvExtensions(e, sourcesOverride); err != nil {
 			sg := m.UI.StepGroup()
-			defer sg.Done()
-			renderStepError(m.UI, sg.Add(""), renderStepReconcile, err)
+			renderStepError(m.UI, sg.Add(""), renderStepReconcileDetect, err)
+			sg.Done()
 			return nil, err
 		}
 
 		log.DebugTitlef("Reconciling environment [%s]", e.Name)
 
 		m.UI.Output(fmt.Sprintf("%s: %s", e.Name, e.File))
 
-		sourcesOverride.diffAndPatch(e.override)
+		if err := sourcesOverride.diffAndPatch(e.override); err != nil {
+			sg := m.UI.StepGroup()
+			renderStepError(m.UI, sg.Add(""), renderStepReconcileApply, err)
+			sg.Done()
+			return nil, err
+		}
 	}
 
 	return m, nil
 }
 
-func validateExtensions(override *composeOverride) error {
-	for _, s := range override.Services {
-		_, err := config.ParseSvcK8sConfigFromMap(s.Extensions)
+func validateEnvExtensions(e *Environment, base *composeOverride) error {
+	for _, s := range e.GetServices() {
+		baseSvc, missingSvcErr := base.getService(s.Name)
+		if missingSvcErr != nil {
+			continue
+		}
+
+		baseSvcK8sCfg, err := config.ParseSvcK8sConfigFromMap(baseSvc.Extensions, config.SkipValidation())
+		if err != nil {
+			return errors.Wrapf(missingSvcErr, "when parsing service %s extensions in base override", baseSvc.Name)
```

Comment:
is it base or override? Am not sure I understand this message correctly.

Reply:
Same as volumes below. It's the service k8s config (via extension) in the base compose file.